common.py

```diff
@@ -524,3 +529,71 @@ def build_sriov_network_operator_check_permissions() -> bool:
     rsh = host.LocalHost()
     ret = rsh.run("podman pull registry.ci.openshift.org/ocp/builder:rhel-9-golang-1.21-openshift-4.16")
     return ret.success()
+
+
+def git_repo_setup(repo_dir: str, *, repo_wipe: bool = True, url: str, branch: str = "master") -> None:
+    exists = os.path.exists(repo_dir)
+    if exists and not repo_wipe:
+        return
+
+    if exists:
+        shutil.rmtree(repo_dir)
+
+    logger.info(f"Cloning repo {url} to {repo_dir}")
+    Repo.clone_from(url, repo_dir, branch=branch)
+
+
+@dataclasses.dataclass
+class ContainerInfo:
```

Comment:
let's make it `@dataclass(frozen=True)`. If you do that, you will need `object.__setattr__(self, "full_tag", ...)` in `__post_init__()`. As suggested at https://docs.python.org/3/library/dataclasses.html#frozen-instances

Reply:
the name is no longer good. It was fine, when it was local at one place. Now the name needs to specify what this is about. It's for `git_build_local()` only, so the name should be somehow reflect that. I'd be fine with an unwielding `GitBuildLocalContainerInfo`.

Or maybe, this anyway should move to a special module ("build_local.py"?). The the module name sufficiently qualifies what this is about, then `ContainerInfo` is good again.

Reply:
What is the advantage of frozen instances?

Reply:
that they are immutable. You get an exception if you try to mutate them, and typing also knows about that (and mypy complains about modifications).

Immutable things are great, because the value is only set at one place (during object construction). The object was in fact immutable already (in the sense, that the current code does not modify an instance). Making it frozen only makes this clearer, with help enforcing that by mypy and runtime checks.